x/reward/keeper/msg_set_reward.go

```diff
@@ -57,25 +57,32 @@ func (k msgServer) SetRewards(goCtx context.Context, msg *types.MsgSetRewards) (
 		}
 	}
 
-	if msg.Coins.Empty() || msg.LastRewardHeight == 0 {
+	if (msg.Coins.Empty() || msg.LastRewardHeight == 0) {
 		rewardPool.InitialCoins = sdk.NewCoins()
 		rewardPool.RemainingCoins = sdk.NewCoins()
 		rewardPool.LastRewardHeight = 0
 		k.RemoveRewardPool(ctx, msg.LaunchID)
+		err = ctx.EventManager().EmitTypedEvent(&types.EventRewardPoolRemoved{LaunchID: msg.LaunchID})
 	} else {
 		rewardPool.InitialCoins = msg.Coins
 		rewardPool.RemainingCoins = msg.Coins
 		rewardPool.Provider = msg.Provider
 		rewardPool.LastRewardHeight = msg.LastRewardHeight
 		k.SetRewardPool(ctx, rewardPool)
+		if !found {
```

Comment:
```suggestion
		if !rewardPoolFound {
```
To keep context since `found` is initialize above